js/onto2ids_v1_0.js

```diff
@@ -0,0 +1,174 @@
+/* SPDX-License-Identifier: Apache-2.0
+   Copyright © 2024 Sander Stolk & Quint Netten */
+
+class Onto2ids_v1_0 {
+
+    /* Transforms input CSV content to buildingSMART IDS in its XML format.
+	   For documentation on the buildinSMART IDS, see: https://github.com/buildingSMART/IDS/ .
+	   
+       @param input: CSV headed by the following column names:	   
+       - ?ontoClassPrefLabel         human-readable name for an ontology class
+       - ?ontoClassURI               unique resource identifier for the ontology class
+       - ?ifcClassLabel              counterpart IFC entity (e.g. IfcDoor)
+       - ?ontoPropertyPrefLabel      human-readable name for an ontology property relevant for the class
+       - ?ontoPropertyURI            unique resource identifier for the ontology property
+       - ?ontoPropertyDatatype       property requirements: datatype
+       - ?ontoPropertyEnumValues     property requirements: enumeration values
+       - ?ontoPropertyCardinalityMin property requirements: min cardinality
+       - ?ontoPropertyCardinalityMax property requirements: max cardinality
+       @param ontoInfo: JSON object with fields 'OrganizationCode', 'DomainName', 'DomainCode', 'DomainVersion'
+	   @param idsInfo: JSON object with fields 'title', 'author', 'date', 'ifcVersion' 
+	   @param idsVersion: Drop down menu for executing different scripts based on the IDS version
+    */
+    static fromCSV(input, ontoInfo, idsInfo) {
+
+        const csvObjects = $.csv.toObjects(input); // see https://github.com/evanplaice/jquery-csv/#documentation		
+		const csvObjectsGrouped = this.groupBy(csvObjects, "ontoClassURI");
+		console.log(JSON.stringify(csvObjectsGrouped));
+
+		let idsXML =
+`<?xml version="1.0" encoding="utf-8"?>
+<ids xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://standards.buildingsmart.org/IDS http://standards.buildingsmart.org/IDS/1.0/ids.xsd" xmlns="http://standards.buildingsmart.org/IDS">
+  
+  <info>
+    <title>${idsInfo.title}</title>
+    <author>${idsInfo.author}</author>
+    <date>${idsInfo.date}</date>
+  </info>
+
+  <specifications>`;
+
+        idsXML += `
+    <specification ifcVersion="${idsInfo.ifcVersion}" name="${ontoInfo.DomainName} - Classification" description="All objects are required to be classified according to the data dictionary ${ontoInfo.DomainName}">
+      <applicability minOccurs="0" maxOccurs="unbounded">
+        <entity>
+          <name>
+            <xs:restriction base="xs:string">`;
+
+			if (idsInfo.ifcVersion == "IFC2X3") {
+				for (const ifcClassName of IFC2X3_CLASSNAMES) {
+					idsXML += `
+					<xs:enumeration value="${ifcClassName}" />`;
+				}
+			} else if (idsInfo.ifcVersion == "IFC4") {
+				for (const ifcClassName of IFC4_CLASSNAMES) {
+					idsXML += `
+					<xs:enumeration value="${ifcClassName}" />`;
+				}
+			} else if (idsInfo.ifcVersion == "IFC4X3_ADD2") {

```

Comment:
Is de IFC4x3_ADD2 waarde aanwezig in het 'idsInfo' JSON object vanaf IDS versie 1.0? Deze wijkt af van eerdere versies, maar mapt nog naar dezelfde IFC4x3_CLASSNAMES lijst.

Reply:
Klopt inderdaad, de benaming van IFC4x3 is aangepast in IDS versie 1.0.
Omdat het dezelfde IFC-versie betreft (met een andere naam), dienen ook dezelfde entiteiten opgeroepen te worden uit de (IFC4x3_CLASSNAMES)